### PR TITLE
CDVD: macOS physical disc support

### PR DESCRIPTION
**Pull request by HarukaMa**
### Description of Changes
Add physical disc support for macOS, moved FreeBSD null code to macOS code and removed conditional compilation from Linux code.

### Rationale behind Changes
For macOS physical disc support.

### Suggested Testing Steps
Test if PCSX2 still builds on FreeBSD.
Test if CDVD module still works on FreeBSD.
Test if physical CDs and DVDs could be played on macOS.

### CODE REVIEW

File: pcsx2/CDVD/Darwin/DriveUtility.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+#endif
+
+std::vector<std::string> GetOpticalDriveList()
+{
+#ifdef __APPLE__
+	std::vector<std::string> drives;
+
+	CFMutableDictionaryRef cd_classes = IOServiceMatching(kIOCDMediaClass);
+	if (cd_classes)
```
Comment (TellowKrinkle) [inline pcsx2/CDVD/Darwin/DriveUtility.cpp:75]:
```suggestion
	if (CFMutableDictionaryRef cd_classes = IOServiceMatching(kIOCDMediaClass))
```

File: pcsx2/CDVD/Darwin/DriveUtility.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	{
+		std::vector<std::string> cd = GetDriveListFromClasses(cd_classes);
+		drives.insert(drives.end(), cd.begin(), cd.end());
+	}
+
+	CFMutableDictionaryRef dvd_classes = IOServiceMatching(kIODVDMediaClass);
+	if (dvd_classes)
```
Comment (TellowKrinkle) [inline pcsx2/CDVD/Darwin/DriveUtility.cpp:82]:
```suggestion
	if (CFMutableDictionaryRef dvd_classes = IOServiceMatching(kIODVDMediaClass))
```

File: pcsx2/CDVD/Darwin/DriveUtility.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	io_iterator_t iterator = IO_OBJECT_NULL;
+	kern_return_t result;
+	std::vector<std::string> drives;
+
+	CFDictionarySetValue(classes, CFSTR(kIOMediaEjectableKey), kCFBooleanTrue);
+	result = IOServiceGetMatchingServices(kIOMasterPortDefault, classes, &iterator);
+	if (result != KERN_SUCCESS)
+		return drives;
+	io_object_t media = IOIteratorNext(iterator);
+	while (media)
```
Comment (TellowKrinkle) [inline pcsx2/CDVD/Darwin/DriveUtility.cpp:45]:
```suggestion
	while (io_object_t media = IOIteratorNext(iterator))
```
and delete the update at the end of the loop

**Comment by lightningterror** [comment]
@HarukaMa Status?

**Review by TellowKrinkle — APPROVED** [review]
Someone else will have to provide freebsd testing

**Comment by lightningterror** [comment]
@martymac can you check freebsd?

**Comment by martymac** [comment]
Hello,

I can't easily access my old physical PS2 discs, however, I can check if everything still builds correctly. I'll try that update ASAP and tell you how it goes!

Cheers.

**Comment by martymac** [comment]
Hello,

Except from the missing `#ifdef` (see the code review), it is OK : the new code is functionally similar to the previous one and it builds fine on FreeBSD :)

Best regards,
Ganael.

**Comment by martymac** [comment]
Hello,

The code has been merged without my previous patch (see my comments). Could someone commit it please ?

Cheers,

Ganael.

**Comment by lightningterror** [comment]
I checked the code and I didn't notice any missing ifdef @martymac , can you specify where?

**Comment by martymac** [comment]
Hello,

In pcsx2/CDVD/Darwin/IOCtlSrc.cpp:69. All code within function IOCtlSrc::SetSpindleSpeed() should be marked ifdef __APPLE__ to build successfully on FreeBSD (see my previous comments).

Thanks !

Ganael.

**Comment by lightningterror** [comment]
A proper solution would be to replace DKIOCDVDSETSPEED DKIOCCDSETSPEED with freebsd alternative or something, but sure as a temporary solution I guess. Maybe some static min safe value would work?

**Comment by HarukaMa** [comment]
Oops, forgot the code still needs to be *compiled* on FreeBSD. Definitely need a `#ifdef` on that function. My bad.
The PR doesn't include FreeBSD physical disc support, so in theory that function won't even be called as the device detection code will just report that there is no physical drive present.

**Comment by lightningterror** [comment]
#8129 should take care of it.

**Comment by martymac** [comment]
Thanks!

**Pull request merged**